### ISSUE TITLE
Update/remove offheap memory settings

### ISSUE THREAD

**Issue by EnriqueL8** [GoogleCloudPlatform/flink-on-k8s-operator#411]
As of Flink 1.11, the properties for setting the offheap memory have changed and no longer match the ones set by the operator. Docs: https://ci.apache.org/projects/flink/flink-docs-release-1.12/deployment/memory/mem_migration.html#taskmanager-heap-size. These properties are always set due to the default webhook setting the `offHeapRatio` and `offHeapMin` fields.

Has there been any discussions to remove the default memory settings since these would be different for each Flink version? Or to update the function to handle multiple versions? IMO harder to handle multiple versions, and easier to let the user configure all the properties directly to match the Flink version they are using.

**Comment by elanv** [comment]
There was a related discussion, but it wasn't going any further and time passed, so it would be nice to discuss it again. https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/issues/288
`spec.taskManager.{memoryOffHeapRatio,memoryOffHeapMin}` and `spec.jobManager.{memoryOffHeapRatio,memoryOffHeapMin}` which are the cause of the problem, automatically set taskmanager.heap.size and jobmanager.heap.size. These were deprecated at 1.10 and 1.11 respectively, but the default value of the FlinkCluster fields interfered with the new memory configurations `*.memory.process.size`.

I would also like to remove the default values of the above fields to eliminate this interference. Instead, it would be nice to clearly guide the CRD document that those fields are only available in versions 1.9 and 1.10 or lower.

However, `*.memory.process.size` is a required value, so if there is memory limit value, it seems necessary to set it to the same value. This configuration is a value that will be ignored in version 1.9 and below, so there should be no problem.

**Comment by EnriqueL8** [comment]
Agreed, doesn't it make more sense to set `*.memmory.process.size` to the memory requested value instead of the memory limit for the container?

**Comment by EnriqueL8** [comment]
Could these environment variables also be deleted? https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/master/controllers/flinkcluster_converter.go#L117-L134

**Comment by elanv** [comment]
When the pod's request value is set, more memory than that value is allowed still, so I wonder if it is appropriate to use the request value to set the total memory usage of Flink.

And I thought it might be better to keep the settings related to off-heap. This is because even if the `*.memory.process.size` is configured, the off heap memory usage used by the flink process will not be completely limited.
